### PR TITLE
feat: added new tracer ElasticAPM

### PR DESCRIPTION
**Pull request by monwolf**
## Related issue
I don't created the issue, but I have the necessity to be able to trace with this APM:  ORY Hydra
## Proposed changes
Added the capability to trace Hydra with Elastic APM
## Checklist

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security
      vulnerability. If this pull request addresses a security vulnerability, I
      confirm that I got green light (please contact
      [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push
      the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation within the code base (if appropriate).

## Further comments

To make it work, you should declare at least ELASTIC_APM_SERVER_URL and  TRACING_PROVIDER=elastic-apm

      # - ELASTIC_APM_SERVER_URL="http://apm-server:8200"
      # - ELASTIC_APM_SERVICE_NAME="ORY Hydra"
      # - ELASTIC_APM_SERVICE_VERSION="1.9.0"
      # - ELASTIC_APM_ENVIRONMENT="devel"

The result should be:

image

### CODE REVIEW

**Review by zepatrik — APPROVED** [review]
Nice one, looks good :+1:

**Review by aeneasr — COMMENTED** [review]
Thank you, this looks great! Would it be possible to add a test for this as well? You can find an example test for zipkin here: https://github.com/ory/x/blob/39005f22c5ce36baf4ae2c179dc3c32309fa0982/tracing/tracer_test.go#L29

**Comment by monwolf** [comment]
@aeneasr  Added the tests

File: tracing/tracer_test.go [hunk 1]
```diff
@@ -68,3 +98,90 @@ func TestZipkinTracer(t *testing.T) {
 		t.Fatalf("Test server did not receive spans")
 	}
 }
+
+func TestElastcApmTracer(t *testing.T) {
+	done := make(chan struct{})
+	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		defer close(done)
+
+		fmt.Println(r.URL)
```
Comment (aeneasr) [inline tracing/tracer_test.go:107]:
```suggestion
```

**Review by aeneasr — APPROVED** [review]
Awesome 🎉 Thank you for your contribution!

File: tracing/tracer_test.go (hunk 1 above)
Comment (monwolf) [reply tracing/tracer_test.go]: 🤦

**Pull request merged**